src/block_disagg/block_disagg_ckpt.c

```diff
@@ -47,7 +47,7 @@ __bmd_checkpoint_pack_raw(WT_BLOCK_DISAGG *block_disagg, WT_SESSION_IMPL *sessio
         WT_RET(__wt_buf_init(session, &ckpt->raw, WT_BLOCK_CHECKPOINT_BUFFER));
         endp = ckpt->raw.mem;
         WT_RET(__wt_block_disagg_write_internal(
-          session, block_disagg, root_image, block_meta, block_meta, &size, &checksum, true, true));
```

Comment:
There's something I'm missing, so I apologise for the potentially dumb question. What's the high-level reason for getting rid of the separate old/new `block_meta` and just having the new?

Reply:
Previously, Peter designed to pass in the old block meta to the function and returns a new block meta after the write. We decide to change that to have the btree own the block meta code. In the new world, btree will be responsible for assigning page ids and checkpoint ids. It will pass in with a block meta with all the new values. The block manager level will fill out the rest missing part in the block meta, such as the lsn.

Reply:
Thanks! The ownership stuff we discussed offline makes this make sense now.